### ISSUE TITLE
Rick Roll Detector (Blacklist-Based)

### ISSUE THREAD

**Issue by tkgo11** [TrianguloY/UrlChecker#508]
### Describe a related problem (optional)

Users are frequently getting tricked by "Rickroll" links in comments, chats, or shared URLs. This disrupts productivity, annoys users, and impacts trust in shared resources.

### Describe your suggested feature

Add a detector that flags or blocks suspicious links known for Rickrolling by comparing URLs against a maintained blacklist. When a match is found, the system should:

Warn the user before opening the link.

Optionally block access and show a custom message.

Allow admins to update or extend the blacklist.

### Describe alternatives you've considered for your suggested feature

Heuristic detection (analyze video titles, etc.) — more complex, may cause false positives.

### Other details

The blacklist includes full YouTube URLs (e.g., https://www.youtube.com/watch?v=dQw4w9WgXcQ)

And... don't take this seriously :)

### Acknowledgements

- [x] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [x] I have written a short but informative title.
- [x] I will fill out all of the requested information in this form.

**Comment by TrianguloY** [comment]
Not sure if joking or not 😅

In any case, I think a full module to detect rickrolls is unnecessary. A more generic system may work, but I think with the current tools you can already detect links.

In fact I have this on my personal patterns list, perhaps I can add it as a default? 🤔

```json
 "Rickroll": {
    "regex": "^https:\/\/www\\.youtube\\.com\/watch\\?v=dQw4w9WgXcQ"
  }
```

The ability to update the patterns based on a external repository is something that was discussed, and I find that useful, but will require changing a good part of the internal format, and honestly I don't think it's currently feasible (I think Pablo already made some progress though)
